### DISCUSSION TITLE
Merge work github account contributions into my personal github account

### DISCUSSION THREAD

**Discussion by orassayag** [community/community#120743] [Other Feature Feedback, Questions, & Ideas]
Hi, I didn't find any clear answer anywhere so I will try my luck here..
I'm writing from my personal GitHub account, while in my day-to-day this account is not really active since I have another account that is related to my work.
Is there any simple way to merge my work account contributions into my personal account?
Thanks.

**Comment by NusRAT-LiA** [comment]


**Reply by MG72021** [reply]
[google.com](url)

**Comment by orassayag** [comment]
@NusRAT-LiA Thanks for the reply, you can assume that my personal and work projects are different, and I don't want to merge between them. I simply want to view in my personal account my work's contribution

**Reply by NusRAT-LiA** [reply]
This merge that I'm talking about is not merging between your personal and work project. It is about forking your work project from your personal project and then merging, so that it becomes your own project ( from personal account ). This will have to effect on your work project and you won't be able to merge any changes from your personal account  to work account. The changes are within the fork.

**Reply by orassayag** [reply]
@NusRAT-LiA  This is OK, but it's not what I meant. I want to see my contribution of my work account in my personal account.

**Reply by dario-macedo-nftStudios** [reply]
@orassayag did you find something useful buddy? I want to do exactly the same.

**Reply by orassayag** [reply]
@dario-macedo-nftStudios No I'm still waiting for some miracle from the GitHub team

**Reply by anmolsahu2k** [reply]
@orassayag please post here if you find anything useful regarding this.

**Comment by mateusbellei** [comment]
you can do this by passing the email that you used on your corporate account to make the commits to your personal account. it should work fine for most cases. it just didn't work for me who made my commits in an ID+Username@users.noreply.github.com email, so far I haven't found an easy or functional way to pass the contributions/commits made in an email of this type

**Comment by night-fury-3** [comment]
Hi orassayag, how are you doing today?

Here is a step-by-step guide on merging contributions from your work account into your personal GitHub account:
**1. Fork the Repository:** Start by forking the repository you wish to transfer contributions from using your work account.
**2. Clone the Forked Repository:** Next, clone the forked repository to your local machine.
**3. Add a Remote:** In your local repository, add a new remote pointing to your personal GitHub account with the following command:
`git remote add personal <URL_TO_PERSONAL_REPO>`
**4. Fetch the Remote Branches:** Fetch the branches from your personal repository by executing:
`git fetch personal`
**5. Checkout Branch:** Switch to the branch where you want to apply the changes in your personal account by running:
`git checkout personal/branch_name`
**6. Cherry-pick Commits:** Select and transfer the desired commits from your work account to your personal account using:
`git cherry-pick <commit_hash>`
**7. Push Changes**: After cherry-picking the commits, push the changes to your personal repository:
`git push personal branch_name`

By following these instructions, you can seamlessly move contributions from your work account to your personal GitHub account.

Please follow the method described above and inform me if it does not work.

**Reply by orassayag** [reply]
While this is perfectly fine, this is not what I meant. I don't want to fork anything or push anything. I want to merge the accounts contributions, the GitHub team should really thing about it.

**Comment by hexfa** [comment]
Hi,

Unfortunately, GitHub doesn’t have a simple way to merge contributions from two accounts into one. Contributions are tied to the email address used for your commits, so even though you're using two different accounts (personal and work), there isn't a built-in feature to combine the contributions into one place while keeping your accounts separate.

Here are a couple of things you can consider:

1. Add Your Work Email to Your Personal Account: You can add the email address associated with your work account to your personal GitHub account. This way, contributions made with that email will appear on your personal contribution graph. However, this might not be ideal since it could expose your work-related contributions on your personal profile, which may not be what you want, especially if your work has policies around this.


2. Keep Them Separate: If you want to keep your personal and work projects entirely separate (as you mentioned), there isn’t a straightforward way to combine contributions on GitHub itself. The only way to fully combine your contributions is to merge the accounts, but that would mix everything, which you don't want.


3. Track Contributions Separately: While GitHub doesn’t support combining them, you could use tools like GitHub Readme Stats to track contributions from both accounts, or manually track your contributions through a portfolio.



Ultimately, GitHub doesn’t currently offer a way to view all contributions across multiple accounts in one place without merging accounts or using a workaround.

You could also try submitting feedback to GitHub to request this feature.

Hope that helps!

**Reply by orassayag** [reply]
How can I suggest them this feature?

**Reply by hexfa** [reply]
> How can I suggest them this feature?

The best and fastest way to do this, I guess is to connect with GitHub seniors through LinkedIn 😁

**Reply by hexfa** [reply]
If my answers are close to your answer, please make it as an answer before closing the discussion 🙏🏻

**Reply by orassayag** [reply]
Unfortunately not :(

**Reply by Maru-Ben** [reply]
Hey i am trying to add the email i use for my work account to my personal one as you stated but i get Email already in use

**Reply by hexfa** [reply]
> Hey i am trying to add the email i use for my work account to my personal one as you stated but i get Email already in use

did you use this email address for GitHub before?

**Comment by VishalKotekar** [comment]
To merge your work GitHub contributions into your personal account, the process isn't directly supported, but you can still see work contributions on your personal profile by ensuring that your work account uses the same email address as your personal account. Contributions tied to the same email address will appear on both accounts. For commits made with a company email, they need to be attributed to the same GitHub email used in your personal account.

**Reply by hexfa** [reply]
Automatic answer from an AI😁

**Reply by orassayag** [reply]
Thanks

**Reply by sh977218** [reply]
in many cases, work email is not same as personal email and cannot be modified. How to merge those two profiles then?

**Comment by crazy-matt** [comment]
I think you are looking for this https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts.

**Comment by felipetodev** [comment]
there is any news? did you find anything @orassayag? 👀

**Reply by orassayag** [reply]
> there is any news? did you find anything @orassayag? 👀

As for this day, no solution was introduced to this issue.

**Comment by AhmadHammad21** [comment]
I'm in the same shoe here..
I don't want to sacrifice 6k contributions xD

**Comment by DaroMacs** [comment]
if you still have access to the github account and the email address related to such account that you want to bring the contributions from, this is the way to do it. It worked for me and it was so easy to achieve.

https://github.com/orgs/community/discussions/23075#discussioncomment-8322359

**Comment by brentengelbrecht** [comment]
The issue is that Github attributes commits to an email address, which is problematic.
A Github user can have multiple GH accounts, and there's no way for one account to have a consolidate "user view" of contributions.
A design flaw in modelling identity that virtually all websites share.
The simple fix is to have a higher-level identifier that consolidates all email addresses owned.
GH seemed to be moving toward that by allowing a user to add different email addresses to an account; however, there are still issues with this.
